drivers/azure/azure.go

```diff
@@ -334,33 +335,65 @@ func (d *Driver) PreCreateCheck() (err error) {
 		}
 	}
 
-	c, err := d.newAzureClient()
+	ctx := context.Background()
+	c, err := d.newAzureClient(ctx)
 	if err != nil {
 		return err
 	}
 
 	// Register used resource providers with current Azure subscription.
-	if err := c.RegisterResourceProviders(
+	if err := c.RegisterResourceProviders(ctx,
 		"Microsoft.Compute",
 		"Microsoft.Network",
-		"Microsoft.Storage"); err != nil {
+		"Microsoft.Storage",
+		"Microsoft.Subscription",
+		"Microsoft.Resources"); err != nil {
 		return err
 	}
 
+	// Validate if the FaultCount and UpdateCount are valid for this location
+	// NOTE(aiyengar2): Imposing a timeout to ensure we don't lag on this call as it is a relatively hefty JSON parse
+	r := make(chan azureutil.FaultDomainCountResponse)
+	faultDomainCountCtx, cancel := context.WithTimeout(ctx, azureutil.MaximumFaultDomainCountTimeout)
+	defer cancel()
+	go func() {
+		c.GetMaximumFaultDomainCount(faultDomainCountCtx, r, d.Location)
+	}()
+	select {
+	case <-faultDomainCountCtx.Done():
+		// Operation timed out
+		log.Warn("Timed out trying to parse the MaximumFaultDomainCount. Continuing execution...")
+	case resp := <-r:
+		if resp.Err != nil {
+			return err
+		} else if d.FaultCount > resp.MaximumFaultDomainCount {
+			return fmt.Errorf("location %s has MaximumPlatformFaultDomainCount=%d, recieved FaultDomainCount=%d",
```

Comment:
typo received

Reply:
Thanks for catching this! Fixed in the latest commit.